swagger.yaml

```diff
@@ -8,6 +8,58 @@ security:
   - bearerAuth: []
 
 paths:
+  "/users/{userId}/vacation":
+    put:
+      operationId: updateUserVacation
+      summary: Update user vacation days
+      description: Update vacation days for a specific user by year.
+      tags:
+        - Update vacation
+      parameters:
+        - name: userId
+          in: path
+          required: true
+          description: The ID of the user to update vacation for
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                vacationDays:
+                  type: object
+                  additionalProperties:
+                    type: object
+                    properties:
+                      total:
+                        type: integer
+                      remaining:
+                        type: integer
+                  description: Vacation days by year
+              required:
+                - vacationDays
+      responses:
+        "200":
+          description: Successfully updated the vacation days
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  id:
+                    type: string
+                    description: The ID of the updated user
+                  updatedFields:
+                    type: array
+                    items:
+                      type: string
+                    description: List of updated fields (e.g., `vacationDaysByYear`, `unspentVacationDaysByYear`, etc.)
+        "400":
+          description: Invalid request, missing required parameters or data
+        "500":
+          description: Server error while updating vacation days
```

Comment:
Don't need to specify the different kind of error statuses, just replace this with the error response that is in use for all the other endpoints